src/main/java/ticTacToe/grid/GridFormatter.java

```diff
@@ -8,35 +8,42 @@
 public class GridFormatter {
 
     private static final String NEW_LINE = "\n";
-    private static final String LINE = "---------";
-    public static final String red_O =("\u001B[31mO\u001b[0m");
-    public static final String blue_X = ("\u001b[34mX\u001b[0m");
+    public static final String red_O = "\u001B[31mO\u001b[0m";
+    public static final String blue_X = "\u001b[34mX\u001b[0m";
 
-    public String prepareGridForPrinting(List<ArrayList<String>> rows) {
+    public String prepareGridForPrinting(List<ArrayList<String>> rows, int gridSize) {
         List<String> printableList = new ArrayList<>();
         printableList.add(NEW_LINE);
-        printableList.add(LINE);
+        printableList.add(line(gridSize));
         printableList.add(NEW_LINE);
         for (List<String> row : rows) {
             printableList.addAll(row);
             printableList.add(NEW_LINE);
-            printableList.add(LINE);
+            printableList.add(line(gridSize));
             printableList.add(NEW_LINE);
         }
-        return String.join(" | ", printableList);
+        return String.join( "| ", printableList);
     }
 
     public List<ArrayList<String>> rowsWithColours(List<ArrayList<String>> rows) {
         for (List<String> row : rows) {
             for (int i = 0; i < row.size(); i++) {
-                String coloredCell = addColorTo(row.get(i));
-                row.set(i, coloredCell);
+                String colouredCell = addColourTo(row.get(i));
+                row.set(i, colouredCell);
             }
         }
         return rows;
     }
 
-    private String addColorTo(String cell) {
+    private String line(int gridSize) {
+        if (gridSize == 3) {
```

Comment:
Can you generalise this? something like: 
```
String line = "";
for (int i = 0; i < gridSize ; i++) {
   line+= "-";
}

return line;
```

Reply:
This way if you onboarded 5x5 it would hopefully still work. As it stands, when you have to check if the board is size 3 or size 4, it points to a violation of the Open Closed Principle.

Reply:
I'm not sure how to do this as it would work for 3x3 grid (it'd add `"-"` up to `gridSize * gridSize`) but it wouldn't work for 4x4 (there are 13 `-` there instead of 16), so I'd have to think about it